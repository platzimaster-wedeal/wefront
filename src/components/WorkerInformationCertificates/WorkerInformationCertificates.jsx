import React from "react";

// Styles
import "../../assets/styles/components/WorkerInformationCertificates/WorkerInformationCertificates.scss";

import Requirements from "../Requirements/Requirements";

const WorkerInformationCertificates = ({ certificates }) => {
 return (
  <div className="worker-information-certificates">
   <strong className="worker-information-certificates__title">
    Experience
   </strong>
   <ul className="worker-information-certificates__list">
    {certificates && certificates.length > 0 ? (
     certificates.map((certificate, id) => (
      <li key={id}>
       {" "}
       <Requirements title={certificate} />{" "}
      </li>
     ))
    ) : (
     <span>This user doesn't have certificates upload</span>
    )}
   </ul>
  </div>
 );
};

export default WorkerInformationCertificates;
