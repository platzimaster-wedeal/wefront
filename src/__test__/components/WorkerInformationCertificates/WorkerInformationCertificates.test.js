import React from "react";
import { mount } from "enzyme";
import { create } from "react-test-renderer";
import WorkerInformationCertificates from "../../../components/WorkerInformationCertificates/WorkerInformationCertificates";

describe("Test for the component WorkerInformationCertificates :D", () => {
 const WorkerInformationCertificatesMounted = mount(
  <WorkerInformationCertificates certificates={["something"]} />
 );
 const WorkerInformationCertificatesCreated = create(
  <WorkerInformationCertificates />
 );

 describe("Visual tests for the component", () => {
  test("should the comopnent have the same tree", () => {
   expect(WorkerInformationCertificatesCreated.toJSON()).toMatchSnapshot();
  });
 });

 describe("Functional tests for the component", () => {
  test("Should the component render one element", () => {
   expect(WorkerInformationCertificatesMounted.length).toEqual(1);
  });
  test("Should the title of the component be a string", () => {
   expect(
    typeof WorkerInformationCertificatesMounted.find(
     ".worker-information-certificates__title"
    ).text()
   ).toEqual("string");
  });
  test("Should the title of the component equal Experience", () => {
   expect(
    WorkerInformationCertificatesMounted.find(
     ".worker-information-certificates__title"
    ).text()
   ).toEqual("Experience");
  });
  test("Should the description of null information be a string", () => {
   expect(
    typeof mount(<WorkerInformationCertificates />)
     .find(".worker-information-certificates__list span")
     .text()
   ).toEqual("string");
  });
  test("Should the description of the component be equal to This user doesn't have certificates upload", () => {
   expect(
    mount(<WorkerInformationCertificates />)
     .find(".worker-information-certificates__list span")
     .text()
   ).toEqual("This user doesn't have certificates upload");
  });
 });
});
