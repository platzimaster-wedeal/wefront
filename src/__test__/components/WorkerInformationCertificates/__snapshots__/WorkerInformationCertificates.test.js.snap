// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test for the component WorkerInformationCertificates :D Visual tests for the component should the comopnent have the same tree 1`] = `
<div
  className="worker-information-certificates"
>
  <strong
    className="worker-information-certificates__title"
  >
    Experience
  </strong>
  <ul
    className="worker-information-certificates__list"
  >
    <span>
      This user doesn't have certificates upload
    </span>
  </ul>
</div>
`;
